spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/ssh/PropertyBasedSshSessionFactory.java

```diff
@@ -193,6 +173,18 @@ protected Iterable<KeyPair> getDefaultKeys(File dir) {
 		return new SingleKeyIdentityProvider(sshKeysByHostname);
 	}
 
+	private static JGitEnvironmentProperties findEnvironmentProperties(
+			Map<String, JGitEnvironmentProperties> sshKeysByHostname, InetSocketAddress socketAddress) {
+
+		JGitEnvironmentProperties sshProperties = sshKeysByHostname.get(socketAddress.getHostString());
+
+		if (sshProperties == null && socketAddress.getAddress() != null) {
+			sshProperties = sshKeysByHostname.get(socketAddress.getAddress().getHostAddress());
```

Comment:
What if this doesn't exist, won't this be null?

Reply:
For every given address,  the map (`sshKeysByHostname`) has either the hostname or the IP address.

The methods in the above classes are called by `JGit` when it wants to establish a connection to the hosts defined in `spring.cloud.config.server.git.uri` or `spring.cloud.config.server.git.repos[].uri`.   Any host in those `URI`s will be added to the Map. So I don't see any scenario where  above method returns null.